### ISSUE TITLE
understanding ktrain under the hood and how to_tfdataset works

### ISSUE THREAD

**Issue by peterbayerle** [amaiya/ktrain#205]
I'm trying to better understand ktrain under the hood, specifically for the classification problem.
The issue I'm running into is understanding to_tfdataset and how that's used to fit the hugging face model.

So given the docs and the labels:
```
x = ['doc1', 'doc2', ...]
y = [1, 0, ...]
y = to_categorical(y)
```
the first step is to make the feature list with a tokenizer `tkzr`:
```
feature_list = []
    for text in x:
        inputs = tkzr( 
            text,
            max_length=max_len,
            truncation=True,
            return_token_type_ids=True,
            return_attention_mask=True
        )
        input_ids, attention_masks, token_type_ids = inputs['input_ids'], inputs['attention_mask'], inputs['token_type_ids']
        feature_list.append([input_ids, attention_masks, token_type_ids])
```
Then in ktrain a tensorflow data set is generated:
```
def gen():
        for idx, data in enumerate(feature_list):
            yield ({'input_ids': data[0],
                     'attention_mask': data[1],
                     'token_type_ids': data[2]},
                    y[idx])

    tfdataset = tf.data.Dataset.from_generator(
        gen,
        ({'input_ids': tf.int32,
          'attention_mask': tf.int32,
          'token_type_ids': tf.int32},
         tf.int64),
        ({'input_ids': tf.TensorShape([None]),
          'attention_mask': tf.TensorShape([None]),
          'token_type_ids': tf.TensorShape([None])},
         tf.TensorShape([None]))
    )
```
From here it should be as simple as creating an optimizer, loss, and compiling/fitting:
```
model.compile(optimizer=optimizer, loss=loss)
model.fit(tfdataset,
          batch_size=BATCH_SIZE,
          epochs=N_EPOCHS)
```
But .... I get an error:
```
W tensorflow/core/kernels/data/generator_dataset_op.cc:103] Error occurred when finalizing GeneratorDataset iterator: Failed precondition: Python interpreter state is not initialized. The process may be terminated.
	 [[{{node PyFunc}}]]
```
This code is essentially taken right from ktrain ... how come it doesn't work?
Apologies if this isn't a great question to be asking here ...
